substrait/common/StringUtils.h

```diff
@@ -0,0 +1,33 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <algorithm>
+#include <iostream>
+#include <string_view>
+
+namespace io::substrait::common {
+class StringUtils final {
+ public:
+  static std::string ltrim(std::string_view s) {
+    size_t start = s.find_first_not_of(WHITESPACE);
+    std::string_view res = (start == std::string::npos) ? "" : s.substr(start);
+    return static_cast<std::string>(res);
+  }
+
+  static std::string rtrim(std::string_view s) {
+    size_t end = s.find_last_not_of(WHITESPACE);
+    std::string_view res = (end == std::string::npos) ? "" : s.substr(0, end + 1);
+    return static_cast<std::string>(res);
+  }
+
+  static std::string trim(std::string_view s) {
+    return rtrim(ltrim(s));
+  }
+
+  static const std::string WHITESPACE;
```

Comment:
```suggestion
  constexpr std::string_view kWhitespace = " \n\r\t\f\v";
```

First, using `string_view` over `string` allows us to make it constexpr and there is no need to separate definition from declaration.

Second, we should be consistent and either use kConstant everywhere or CONSTANT everywhere and we are using kConstant elsewhere (e.g. Exceptions.h)